c/example04.c

```diff
@@ -0,0 +1,148 @@
+/* Example 04 :
+This example shows how to create and delete a tag for a specified resource.
+*/
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <curl/curl.h>
+
+#define AK_SIZE 20
+#define SK_SIZE 40
+
+#define auto_tag  __attribute__ ((__cleanup__(delete_tag_cleanup)))
+
+struct Tag {
+    char *resource_id;
+    char *key;
+    char *value;
+    CURL *curl;
+    struct curl_slist *headers;
+};
+
+
+/* Function to initialize the headers */
+void init_headers(struct curl_slist **headers)
+{
+    *headers = curl_slist_append(*headers, "Content-Type: application/json");
```

Comment:
I'm not convince, that making a function for only 1 line of code, is very useful.
I mean, 
```
*headers = curl_slist_append(*headers, "Content-Type: application/json");
```
this tell exactely what you are doing: adding Content-Type header to headers list, and setting it to "application/json".

where, when you read `init_headers()`, you know you init header, but don't know how. and force you to jump to this function in order to understand this. it's juste more time consuming to understand this.